tests/countries/test_benin.py

```diff
@@ -0,0 +1,271 @@
+#  holidays
+#  --------
+#  A fast, efficient Python library for generating country, province and state
+#  specific sets of holidays on the fly. It aims to make determining whether a
+#  specific date is a holiday as fast and flexible as possible.
+#
+#  Authors: Vacanza Team and individual contributors (see CONTRIBUTORS file)
+#           dr-prodigy <dr.prodigy.github@gmail.com> (c) 2017-2023
+#           ryanss <ryanssdev@icloud.com> (c) 2014-2017
+#  Website: https://github.com/vacanza/holidays
+#  License: MIT (see LICENSE file)
+
+from unittest import TestCase
+
+from holidays.constants import WORKDAY
+from holidays.countries.benin import Benin, BJ, BEN
+from tests.common import CommonCountryTests
+
+
+class TestBenin(CommonCountryTests, TestCase):
+    @classmethod
+    def setUpClass(cls):
+        years = range(1977, 2050)
+        super().setUpClass(Benin, years=years)
+        cls.no_estimated_holidays = Benin(years=years, islamic_show_estimated=False)
+
+    def test_country_aliases(self):
+        self.assertAliases(Benin, BJ, BEN)
+
+    def test_no_holidays(self):
+        self.assertNoHolidays(Benin(years=1976))
+
+    def test_new_years_day(self):
+        self.assertHolidayName(
+            "Fête du Nouvel An", (f"{year}-01-01" for year in range(1977, 2050))
+        )
+
+    def test_vodoun_festival(self):
+        name = "Fête annuelle des religions traditionnelles"
+        self.assertHolidayName(name, (f"{year}-01-10" for year in range(1998, 2025)))
+        self.assertHolidayName(
+            name,
+            "2025-01-09",
+            "2025-01-10",
+        )
+
+    def test_martyrs_day(self):
+        name = "Journée des Martyrs"
+        self.assertHolidayName(name, (f"{year}-01-16" for year in range(1980, 1990)))
+        self.assertNoHolidayName(name, range(1977, 1979), range(1991, 2050))
```

Comment:
```suggestion
        self.assertHolidayName(name, (f"{year}-01-16" for year in range(1980, 1991)))
        self.assertNoHolidayName(name, range(1977, 1980), range(1991, 2050))
```
And same for similar range cases below.